### DISCUSSION TITLE
[Feature] Android App - Trust user installed CA certs

### DISCUSSION THREAD

**Discussion by d-sko** [immich-app/immich#2139] [Feature Request]
### The feature

It would be great if the Android app would trust user installed CA certs, so that it would communicate over TLS with servers that use certificates from an internal selfhosted CA. This was the default until API level 23. Since API 24 it needs to be allowed in the AndroidManifest.xml, see last example in base-config: https://developer.android.com/training/articles/security-config#base-config

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
We're using Dart, which has its own certificate store. We've looked into this previously and couldn't find a way to add new certificates to that.

With ACME and DNS-01 challenges, it's pretty easy these days to get a "real" certificate for free without needing to expose your services at all. I'd recommend that approach instead of using self-signed certs.

**Reply by ddshd** [reply]
Update: PR #4051 allows bypassing SSL cert check for the immich server

**Comment by d-sko** [comment]
Sorry, I didn't know that Dart makes this more complex. I know about DNS-01, its just a bit more complex to setup for automatic updates because the acme client needs to update the challenge-DNS-record, which requires a registrar with an API for updates. Currently my Traefik just gets the cert from my internal Step-CA with HTTP challenge.
But I'll look in the DNS-01-Setup with my external domain, should be doable, thanks!

**Comment by Santobert** [comment]
Would it be an option to allow the user to bypass the verification? I would prefer encrypted but unverified traffic to complete lack of encryption. The best option would of course be to use the certificates installed in the system, but I understand that this might be difficult to achieve.

**Reply by Santobert** [reply]
Maybe relevant: https://github.com/dart-lang/sdk/issues/50435

**Reply by ddshd** [reply]
PR #4051 allows bypassing the SSL cert check
